### PR TITLE
IN Sockets for Messaging

### PR DESCRIPTION
**Pull request by HAZberi**
### What this PR does (required):
- Integrate Sockets for real time messaging.

### Screenshots / Videos (required):

https://user-images.githubusercontent.com/72773994/127581239-769ebfe6-a712-4394-90c0-6af241d0d8d1.mov

### Any information needed to test this feature (required):
- Log in with two different accounts on separate windows. Start sending messages between accounts. You should expect a similar functionality as shown in the attached video.

### Any issues with the current functionality (optional):
- Nothing, working as expected.

### CODE REVIEW

**Review by rajivtitus — COMMENTED** [review]
Hey @HAZberi, if you've based this PR off another branch then please adjust the base of this PR so that only new changes are reflected.

**Comment by HAZberi** [comment]
Please review the sockets integration for messaging. Thank you

**Review by rajivtitus — APPROVED** [review]
Good work with this!

File: client/src/pages/Messages/Messages.tsx [hunk 1]
```diff
@@ -28,6 +32,11 @@ export default function Messages(): JSX.Element {
       setSearch('');
     }
   };
+  useEffect(() => {
```
Comment (rajivtitus) [inline client/src/pages/Messages/Messages.tsx]: Maybe add this to the socket context? :)

File: client/src/pages/Messages/Messages.tsx [hunk 2]
```diff
@@ -28,6 +32,11 @@ export default function Messages(): JSX.Element {
...
+    if (socket) {
+      socket.emit('addUser', loggedInUser?.id);
```
Comment (rajivtitus) [inline client/src/pages/Messages/Messages.tsx]: Maybe also check if a logged in user is available here?

File: client/src/pages/Messages/Messages.tsx (hunk 1 above)
Comment (HAZberi) [reply client/src/pages/Messages/Messages.tsx]: Successfully moved this useEffect to socket context. 

commit 5f5ee0e5013612f4bc4fc708bbc8309c80820c4e

File: client/src/pages/Messages/Messages.tsx (hunk 2 above)
Comment (HAZberi) [reply client/src/pages/Messages/Messages.tsx]: Successfully added suggested if case. 

commit 5f5ee0e5013612f4bc4fc708bbc8309c80820c4e

**Comment by HAZberi** [comment]
Successfully implemented request changes.

**Pull request merged**